ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandRunnerTest.java

```diff
@@ -16,80 +16,90 @@
 
-    CommandRunner commandRunner = new CommandRunner(statementExecutor, new MockCommandStore
-        ("CT", commandConsumer, null,
-         new CommandIdAssigner(new MetaStoreImpl())));
-    return commandRunner;
+    return statementExecutor;
   }
 
-
   @Test
-  public void testThread() throws InterruptedException {
-    MockKsqkEngine mockKsqkEngine = new MockKsqkEngine(
-        TestUtils.getMockKsqlConfig(), new MockKafkaTopicClient());
-    StatementParser statementParser = new StatementParser(mockKsqkEngine);
-    StatementExecutor statementExecutor = new StatementExecutor(mockKsqkEngine, statementParser);
+  public void testNewCommandRun() throws Exception {
+    StatementExecutor statementExecutor = getMockStatementExecutor();
+    statementExecutor.handleStatement(anyObject(), anyObject());
+    expectLastCall().times(4);
+    replay(statementExecutor);
     CommandRunner commandRunner = getCommanRunner(statementExecutor);
-    new Thread(commandRunner).start();
-    Thread.sleep(50000);
-    commandRunner.close();
+    commandRunner.fetchAndRunCommands();
     CommandId topicCommandId =  new CommandId(CommandId.Type.TOPIC, "_CSASTopicGen");
     CommandId csCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASStreamGen");
     CommandId csasCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASGen");
     CommandId ctasCommandId =  new CommandId(CommandId.Type.TABLE, "_CTASGen");
 
     Map<CommandId, CommandStatus> statusStore = statementExecutor.getStatuses();
     Assert.assertNotNull(statusStore);
-//    Assert.assertEquals(4, statusStore.size());
     Assert.assertEquals(CommandStatus.Status.SUCCESS, statusStore.get(topicCommandId).getStatus());
     Assert.assertEquals(CommandStatus.Status.SUCCESS, statusStore.get(csCommandId).getStatus());
-//    Assert.assertEquals(CommandStatus.Status.SUCCESS, statusStore.get(csasCommandId).getStatus());
-//    Assert.assertEquals(CommandStatus.Status.ERROR, statusStore.get(ctasCommandId).getStatus());
+    Assert.assertEquals(CommandStatus.Status.SUCCESS, statusStore.get(csasCommandId).getStatus());
+    Assert.assertEquals(CommandStatus.Status.ERROR, statusStore.get(ctasCommandId).getStatus());
+    verify(statementExecutor);
   }
 
   @Test
   public void testPriorCommandsRun() throws Exception {
-    MockKsqkEngine mockKsqkEngine = new MockKsqkEngine(
-        TestUtils.getMockKsqlConfig(), new MockKafkaTopicClient());
-    StatementParser statementParser = new StatementParser(mockKsqkEngine);
-    StatementExecutor statementExecutor = new StatementExecutor(mockKsqkEngine, statementParser);
+    StatementExecutor statementExecutor = getMockStatementExecutor();
+    statementExecutor.handleStatements(anyObject());
+    expectLastCall();
+    replay(statementExecutor);
     CommandRunner commandRunner = getCommanRunner(statementExecutor);
     commandRunner.processPriorCommands();
     CommandId topicCommandId =  new CommandId(CommandId.Type.TOPIC, "_CSASTopicGen");
```

Comment:
Ditto here: all this is not needed as it is testing the mock `StatementExecutor` not the `CommandRunner`

Reply:
Refactored as mentioned above.